giskard/integrations/avid.py

```diff
@@ -47,8 +47,9 @@ def create_report_from_issue(issue: Issue, model: BaseModel, dataset: Dataset =
         AVID Report created from the issue.
     """
     # Prepare artifacts
-    artifacts = [Artifact(type=ArtifactTypeEnum.model, name=model.meta.name)]
-    if dataset:
+    model_name = model.meta.name or model.__class__.__name__
```

Comment:
Your previous implementation was fine. This is already done for you in the BaseModel class. I'd leave it as it was.